In [1]:
from ultralytics import YOLO

In [2]:
from ultralytics import settings
settings.update({"datasets_dir": "/home/tj_students/leaf-localisation-main"})

In [3]:
model = YOLO("models/best_m.pt")

In [4]:
metrics = model.val(data="../data.yaml", imgsz=1280)
print(f"Precision  : {metrics.box.mp:.3f}")
print(f"Recall     : {metrics.box.mr:.3f}")
print(f"mAP50      : {metrics.box.map50:.3f}")
print(f"mAP50-95   : {metrics.box.map:.3f}")

Ultralytics 8.4.104 🚀 Python-3.10.12 torch-2.13.0+cu130 CUDA:0 (NVIDIA RTX A5000, 24111MiB)


YOLO26m summary (fused): 132 layers, 20,350,223 parameters, 0 gradients, 67.8 GFLOPs


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 5668.7±2841.0 MB/s, size: 609.4 KB)


val: Scanning /home/tj_students/leaf-localisation-main/data/processed/val/labels.cache... 567 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 567/567 132.1Mit/s 0.0s

val: /home/tj_students/leaf-localisation-main/data/processed/val/images/soycotton_2b47c43d-2fed-4e06-ace5-47c6744417ad.jpeg: 1 duplicate labels removed
val: /home/tj_students/leaf-localisation-main/data/processed/val/images/soycotton_38285a73-29ba-4efd-bf1b-73b6ad7eaf94.jpeg: 1 duplicate labels removed


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 1/36 1.3s/it 0.4s<44.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 2/36 1.2it/s 0.9s<29.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ━─────────── 3/36 1.4it/s 1.4s<23.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 4/36 1.7it/s 1.8s<18.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 5/36 2.2it/s 2.1s<14.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━━────────── 6/36 2.7it/s 2.3s<11.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 7/36 2.9it/s 2.6s<9.9s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 8/36 3.1it/s 2.9s<9.0s

[W807 15:12:12.054301897 CUDACachingAllocator.cpp:3933] memory allocation failed with OOM on device 0 while trying to allocate 1335885824 bytes (free: 1357381632, total: 25282478080).


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 9/36 3.0it/s 3.3s<9.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 10/36 3.0it/s 3.6s<8.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 11/36 3.1it/s 3.9s<8.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━━──────── 12/36 3.2it/s 4.2s<7.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 13/36 3.1it/s 4.6s<7.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 14/36 3.2it/s 4.8s<6.9s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━━─────── 15/36 3.2it/s 5.1s<6.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 16/36 3.3it/s 5.4s<6.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 17/36 3.3it/s 5.7s<5.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 18/36 3.3it/s 6.0s<5.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 19/36 3.3it/s 6.3s<5.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 20/36 3.4it/s 6.6s<4.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━━───── 21/36 3.3it/s 7.0s<4.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 22/36 3.2it/s 7.3s<4.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 23/36 3.0it/s 7.7s<4.3s

[W807 15:12:17.794844429 CUDACachingAllocator.cpp:3933] memory allocation failed with OOM on device 0 while trying to allocate 1763704832 bytes (free: 795344896, total: 25282478080).


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━━──── 24/36 2.7it/s 8.2s<4.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 25/36 2.7it/s 8.6s<4.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 26/36 2.7it/s 9.0s<3.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 27/36 2.8it/s 9.3s<3.2s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 28/36 2.9it/s 9.6s<2.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 29/36 2.9it/s 10.0s<2.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━━── 30/36 3.0it/s 10.3s<2.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 31/36 3.1it/s 10.6s<1.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 32/36 3.1it/s 10.9s<1.3s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━━─ 33/36 3.1it/s 11.2s<1.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 34/36 3.2it/s 11.5s<0.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 35/36 3.3it/s 11.8s<0.3s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 36/36 3.0it/s 11.9s

                   all        567       4129      0.803      0.772      0.854      0.688


Speed: 1.9ms preprocess, 16.3ms inference, 0.0ms loss, 0.1ms postprocess per image


Results saved to /home/tj_students/leaf-localisation-main/deployment/runs/detect/val-26


Precision  : 0.803
Recall     : 0.772
mAP50      : 0.854
mAP50-95   : 0.688


## INT8 export for Coral Edge TPU + Raspberry Pi

**What we tried and why it changed:**

- `format="tflite"` is a deprecated alias in this Ultralytics version that silently redirects to `format="litert"` — the notebook's original two export cells were really the same thing twice, and both passed invalid quantize arguments (`quantize="int8"` as a string, `int8=True`) that this API doesn't recognize (the real value is the integer `quantize=8`) — silently falling back to plain FP32 with no size benefit.
- `format="edgetpu"` (the ONNX → `onnx2tf` → TensorFlow → TFLite pipeline) is Ultralytics' built-in path for this, but on this model it hangs indefinitely inside TensorFlow's own converter — reproduced 5+ times across different `onnx2tf`/TensorFlow versions, with plain FP32 conversion (no quantization at all) hanging identically. This looks like a genuine `onnx2tf`/TensorFlow compatibility gap with the very new YOLO26 architecture, not something fixable by version-pinning.
- **The fix: `format="litert"`.** This is Ultralytics' officially-documented, Coral-compatible export path, and it uses a completely different, PyTorch-native pipeline (`litert_torch`, via `torch.export`) that never touches `onnx2tf`/TensorFlow at all — so it sidesteps the hang entirely. Verified working: exports complete in a bounded, predictable time (~13 min at imgsz=640, ~43 min at imgsz=1280 for full INT8; the weight-only mode below takes ~20s regardless of size).
- **The catch:** `litert_torch` requires `torch<2.13.0`, but this project's training environment uses `torch==2.13.0+cu130`. Installing it directly here once already force-downgraded torch and broke `torchvision` — so **export must run in a separate virtual environment**, not this notebook's kernel. The cell below creates one (`export_venv/`, one-time, ~10 min) and shells out to it for export only; validation still runs in this kernel since reading a `.tflite` file doesn't need torch at all.

**The accuracy trade-off (measured at imgsz=1280, the production resolution):**

| Model | Precision | Recall | mAP50 | mAP50-95 |
|---|---|---|---|---|
| PyTorch baseline (fp32) | 0.803 | 0.772 | 0.854 | **0.688** |
| `quantize=8` full INT8 (**required for the physical Coral TPU chip**) | 0.535 | 0.777 | 0.604 | **0.344** (‑50%) |
| `quantize='w8a32'` weight-only INT8 (Raspberry Pi CPU only, **not** Coral-compatible) | 0.795 | 0.765 | 0.846 | **0.679** (‑1.3%) |

Full INT8 — the only mode the Edge TPU chip can actually execute — roughly **halves mAP50-95** on this model (a known YOLO/SiLU-activation quantization weakness, confirmed here at both imgsz=640 and imgsz=1280, so it isn't a resolution artifact). Weight-only INT8 keeps accuracy essentially intact (mAP50-95 0.688→0.679) and is still ~3.8x smaller than fp32, but it cannot run on the Coral chip itself — it's a CPU-only model, good for Raspberry Pi alone.

**Given accuracy is the priority: the cells below default to `quantize='w8a32'` for Raspberry Pi.** A full-INT8 Coral export cell is included but commented out — only uncomment it if you've decided the ~50% mAP50-95 loss is an acceptable trade for running on the physical TPU (or once the model has been fine-tuned/QAT-trained to recover accuracy under INT8).

In [5]:
# One-time setup: a dedicated venv for export only (needs torch<2.13, which would conflict
# with this project's training torch==2.13.0 if installed in the same environment). Safe to re-run.
!python3 -m venv export_venv
!export_venv/bin/pip install -q --upgrade pip
!export_venv/bin/pip install -q "torch==2.12.1" "torchvision==0.27.1" --index-url https://download.pytorch.org/whl/cpu
!export_venv/bin/pip install -q "ultralytics==8.4.104" "litert-torch>=0.9.0" "ai-edge-litert>=2.1.4" "ai-edge-quantizer>=0.6.0" opencv-python-headless nvidia-ml-py ultralytics-thop

In [6]:
# Weight-only INT8 export (Raspberry Pi CPU, accuracy-preserving, ~4x smaller than fp32).
# Runs in export_venv (not this kernel) since litert_torch needs torch<2.13.
!export_venv/bin/python -c "from ultralytics import YOLO; YOLO('models/best_m.pt').export(format='litert', quantize='w8a32', imgsz=1280)"

# Full INT8 export for the physical Coral Edge TPU chip: uncomment only after accepting the
# ~50% mAP50-95 loss measured above (or after fine-tuning/QAT to recover accuracy under INT8).
# Takes ~40 min at imgsz=1280 (calibrates over the full validation set from data.yaml).
# !export_venv/bin/python -c "from ultralytics import YOLO; YOLO('models/best_m.pt').export(format='litert', quantize=8, data='../data.yaml', imgsz=1280)"

int8_tflite_path = "models/best_m_w8a32.tflite"  # switch to "models/best_m_int8.tflite" if using the full-INT8 cell above
print("INT8 TFLite path:", int8_tflite_path)

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1075, in _handle_fromlist
  File "/home/tj_students/leaf-localisation-main/deployment/export_venv/lib/python3.10/site-packages/ultralytics/__init__.py", line 38, in __getattr__
    return getattr(importlib.import_module("ultralytics.models"), name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/tj_students/leaf-localisation-main/deployment/export_venv/lib/python3.10/site-packages/ultralytics/models/__init__.py", line 3, in <module>
    from .fastsam import FastSAM
  File "/home/tj_students/leaf-localisation-main/deployment/export_venv/lib/python3.10/site-packages/ultralytics/models/fastsam/__init__.py", line 3, in <module>
    from .model import FastSAM
  File "/home/tj_students/leaf-localisation-main/deployment/export_venv/lib/python3.10/site-packages/ultralyt

INT8 TFLite path: models/best_m_w8a32.tflite


In [7]:
int8_metrics = YOLO(int8_tflite_path).val(data="../data.yaml", imgsz=1280)
print(f"Precision  : {int8_metrics.box.mp:.3f}")
print(f"Recall     : {int8_metrics.box.mr:.3f}")
print(f"mAP50      : {int8_metrics.box.map50:.3f}")
print(f"mAP50-95   : {int8_metrics.box.map:.3f}")

WARNING ⚠️ Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify', 'pose', 'obb' or 'semantic'.


Ultralytics 8.4.104 🚀 Python-3.10.12 torch-2.13.0+cu130 CUDA:0 (NVIDIA RTX A5000, 24111MiB)


Loading models/best_m_w8a32.tflite for LiteRT inference...


Setting batch=1 input of shape (1, 3, 1280, 1280)


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1498.2±370.1 MB/s, size: 587.9 KB)


val: Scanning /home/tj_students/leaf-localisation-main/data/processed/val/labels.cache... 567 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 567/567 79.3Mit/s 0.0s

val: /home/tj_students/leaf-localisation-main/data/processed/val/images/soycotton_2b47c43d-2fed-4e06-ace5-47c6744417ad.jpeg: 1 duplicate labels removed
val: /home/tj_students/leaf-localisation-main/data/processed/val/images/soycotton_38285a73-29ba-4efd-bf1b-73b6ad7eaf94.jpeg: 1 duplicate labels removed


INFO: Created TensorFlow Lite XNNPACK delegate for CPU.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 0% ──────────── 1/567 9.4s/it 2.8s<1:28:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 0% ──────────── 2/567 5.5s/it 5.6s<51:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 0% ──────────── 3/567 4.2s/it 8.3s<39:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 0% ──────────── 4/567 3.6s/it 11.1s<34:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 0% ──────────── 5/567 3.3s/it 13.8s<31:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 6/567 3.1s/it 16.6s<29:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 7/567 3.0s/it 19.3s<27:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 8/567 2.9s/it 22.0s<27:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 9/567 2.8s/it 24.8s<26:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 10/567 2.8s/it 27.5s<26:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 1% ──────────── 11/567 2.8s/it 30.2s<25:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 12/567 2.8s/it 32.9s<25:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 13/567 2.8s/it 35.6s<25:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 14/567 2.7s/it 38.4s<25:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 15/567 2.7s/it 41.1s<25:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 16/567 2.7s/it 43.8s<25:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 2% ──────────── 17/567 2.7s/it 46.5s<25:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 3% ──────────── 18/567 2.7s/it 49.2s<24:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 3% ──────────── 19/567 2.7s/it 52.0s<24:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 3% ──────────── 20/567 2.7s/it 54.7s<24:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 3% ──────────── 21/567 2.7s/it 57.4s<24:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 3% ──────────── 22/567 2.7s/it 1:00<24:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ──────────── 23/567 2.7s/it 1:03<24:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ╸─────────── 24/567 2.7s/it 1:06<24:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ╸─────────── 25/567 2.7s/it 1:08<24:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ╸─────────── 26/567 2.7s/it 1:11<24:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ╸─────────── 27/567 2.7s/it 1:14<24:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 4% ╸─────────── 28/567 2.7s/it 1:17<24:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 29/567 2.7s/it 1:19<24:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 30/567 2.7s/it 1:22<24:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 31/567 2.7s/it 1:25<24:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 32/567 2.7s/it 1:28<24:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 33/567 2.7s/it 1:30<24:26

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 5% ╸─────────── 34/567 2.7s/it 1:33<24:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 6% ╸─────────── 35/567 2.7s/it 1:36<24:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 6% ╸─────────── 36/567 2.7s/it 1:39<24:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 6% ╸─────────── 37/567 2.7s/it 1:41<24:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 6% ╸─────────── 38/567 2.7s/it 1:44<24:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 6% ╸─────────── 39/567 2.7s/it 1:47<24:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 40/567 2.7s/it 1:50<24:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 41/567 2.7s/it 1:52<24:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 42/567 2.7s/it 1:55<24:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 43/567 2.7s/it 1:58<23:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 44/567 2.7s/it 2:01<23:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 7% ╸─────────── 45/567 2.7s/it 2:03<23:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ╸─────────── 46/567 2.7s/it 2:06<23:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ╸─────────── 47/567 2.7s/it 2:09<23:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ━─────────── 48/567 2.7s/it 2:12<23:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ━─────────── 49/567 2.8s/it 2:14<23:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ━─────────── 50/567 2.8s/it 2:17<23:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 8% ━─────────── 51/567 2.8s/it 2:20<23:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 9% ━─────────── 52/567 2.7s/it 2:23<23:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 9% ━─────────── 53/567 2.8s/it 2:25<23:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 9% ━─────────── 54/567 2.8s/it 2:28<23:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 9% ━─────────── 55/567 2.8s/it 2:31<23:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 9% ━─────────── 56/567 2.8s/it 2:34<23:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 57/567 2.7s/it 2:36<23:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 58/567 2.7s/it 2:39<23:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 59/567 2.7s/it 2:42<23:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 60/567 2.7s/it 2:45<23:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 61/567 2.7s/it 2:47<23:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 10% ━─────────── 62/567 2.7s/it 2:50<23:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 63/567 2.7s/it 2:53<23:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 64/567 2.7s/it 2:56<23:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 65/567 2.7s/it 2:58<22:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 66/567 2.7s/it 3:01<22:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 67/567 2.8s/it 3:04<22:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 11% ━─────────── 68/567 2.7s/it 3:07<22:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 12% ━─────────── 69/567 2.7s/it 3:09<22:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 12% ━─────────── 70/567 2.8s/it 3:12<22:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 12% ━╸────────── 71/567 2.8s/it 3:15<22:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 12% ━╸────────── 72/567 2.8s/it 3:18<22:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 12% ━╸────────── 73/567 2.8s/it 3:20<22:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 74/567 2.7s/it 3:23<22:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 75/567 2.8s/it 3:26<22:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 76/567 2.8s/it 3:29<22:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 77/567 2.8s/it 3:31<22:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 78/567 2.8s/it 3:34<22:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 13% ━╸────────── 79/567 2.8s/it 3:37<22:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 80/567 2.8s/it 3:40<22:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 81/567 2.7s/it 3:42<22:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 82/567 2.7s/it 3:45<22:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 83/567 2.7s/it 3:48<22:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 84/567 2.7s/it 3:51<22:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 14% ━╸────────── 85/567 2.8s/it 3:53<22:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 15% ━╸────────── 86/567 2.8s/it 3:56<22:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 15% ━╸────────── 87/567 2.8s/it 3:59<22:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 15% ━╸────────── 88/567 2.7s/it 4:02<21:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 15% ━╸────────── 89/567 2.7s/it 4:04<21:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 15% ━╸────────── 90/567 2.7s/it 4:07<21:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━╸────────── 91/567 2.7s/it 4:10<21:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━╸────────── 92/567 2.7s/it 4:13<21:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━╸────────── 93/567 2.7s/it 4:15<21:41

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━╸────────── 94/567 2.7s/it 4:18<21:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━━────────── 95/567 2.7s/it 4:21<21:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 16% ━━────────── 96/567 2.7s/it 4:24<21:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 97/567 2.7s/it 4:26<21:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 98/567 2.7s/it 4:29<21:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 99/567 2.7s/it 4:32<21:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 100/567 2.7s/it 4:35<21:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 101/567 2.7s/it 4:37<21:20

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 17% ━━────────── 102/567 2.7s/it 4:40<21:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 18% ━━────────── 103/567 2.7s/it 4:43<21:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 18% ━━────────── 104/567 2.7s/it 4:46<21:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 18% ━━────────── 105/567 2.7s/it 4:48<21:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 18% ━━────────── 106/567 2.7s/it 4:51<21:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 18% ━━────────── 107/567 2.7s/it 4:54<21:00

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 108/567 2.7s/it 4:57<20:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 109/567 2.7s/it 4:59<20:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 110/567 2.7s/it 5:02<20:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 111/567 2.7s/it 5:05<20:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 112/567 2.7s/it 5:07<20:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 19% ━━────────── 113/567 2.7s/it 5:10<20:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━────────── 114/567 2.7s/it 5:13<20:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━────────── 115/567 2.7s/it 5:16<20:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━────────── 116/567 2.7s/it 5:18<20:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━────────── 117/567 2.7s/it 5:21<20:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━────────── 118/567 2.7s/it 5:24<20:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 20% ━━╸───────── 119/567 2.7s/it 5:27<20:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 21% ━━╸───────── 120/567 2.7s/it 5:29<20:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 21% ━━╸───────── 121/567 2.7s/it 5:32<20:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 21% ━━╸───────── 122/567 2.7s/it 5:35<20:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 21% ━━╸───────── 123/567 2.7s/it 5:38<20:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 21% ━━╸───────── 124/567 2.7s/it 5:40<20:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 125/567 2.7s/it 5:43<20:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 126/567 2.7s/it 5:46<20:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 127/567 2.7s/it 5:49<20:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 128/567 2.7s/it 5:51<20:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 129/567 2.7s/it 5:54<19:60

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 22% ━━╸───────── 130/567 2.7s/it 5:57<19:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 131/567 2.7s/it 5:60<19:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 132/567 2.7s/it 6:02<19:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 133/567 2.7s/it 6:05<19:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 134/567 2.7s/it 6:08<19:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 135/567 2.7s/it 6:10<19:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 23% ━━╸───────── 136/567 2.7s/it 6:13<19:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 24% ━━╸───────── 137/567 2.7s/it 6:16<19:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 24% ━━╸───────── 138/567 2.7s/it 6:19<19:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 24% ━━╸───────── 139/567 2.7s/it 6:21<19:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 24% ━━╸───────── 140/567 2.7s/it 6:24<19:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 24% ━━╸───────── 141/567 2.7s/it 6:27<19:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 142/567 2.7s/it 6:29<19:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 143/567 2.7s/it 6:32<19:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 144/567 2.7s/it 6:35<19:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 145/567 2.7s/it 6:38<19:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 146/567 2.7s/it 6:40<19:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 25% ━━━───────── 147/567 2.7s/it 6:43<19:00

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 148/567 2.7s/it 6:46<19:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 149/567 2.7s/it 6:49<18:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 150/567 2.7s/it 6:51<18:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 151/567 2.7s/it 6:54<18:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 152/567 2.7s/it 6:57<18:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 26% ━━━───────── 153/567 2.7s/it 6:59<18:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 154/567 2.7s/it 7:02<18:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 155/567 2.7s/it 7:05<18:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 156/567 2.7s/it 7:08<18:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 157/567 2.7s/it 7:10<18:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 27% ━━━───────── 158/567 2.7s/it 7:13<18:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 159/567 2.7s/it 7:16<18:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 160/567 2.7s/it 7:18<18:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 161/567 2.7s/it 7:21<18:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 162/567 2.7s/it 7:24<18:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 163/567 2.7s/it 7:26<18:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 28% ━━━───────── 164/567 2.7s/it 7:29<18:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━───────── 165/567 2.7s/it 7:32<18:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━╸──────── 166/567 2.7s/it 7:35<18:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━╸──────── 167/567 2.7s/it 7:37<18:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━╸──────── 168/567 2.7s/it 7:40<18:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━╸──────── 169/567 2.7s/it 7:43<18:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 29% ━━━╸──────── 170/567 2.7s/it 7:46<18:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 171/567 2.8s/it 7:48<18:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 172/567 2.7s/it 7:51<18:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 173/567 2.7s/it 7:54<17:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 174/567 2.7s/it 7:57<17:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 30% ━━━╸──────── 175/567 2.7s/it 7:59<17:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 176/567 2.7s/it 8:02<17:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 177/567 2.7s/it 8:05<17:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 178/567 2.7s/it 8:07<17:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 179/567 2.7s/it 8:10<17:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 180/567 2.7s/it 8:13<17:41

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 31% ━━━╸──────── 181/567 2.7s/it 8:16<17:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 182/567 2.7s/it 8:18<17:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 183/567 2.7s/it 8:21<17:26

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 184/567 2.7s/it 8:24<17:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 185/567 2.7s/it 8:27<17:20

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 186/567 2.7s/it 8:29<17:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 32% ━━━╸──────── 187/567 2.7s/it 8:32<17:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━╸──────── 188/567 2.7s/it 8:35<17:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━━──────── 189/567 2.7s/it 8:37<17:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━━──────── 190/567 2.7s/it 8:40<17:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━━──────── 191/567 2.7s/it 8:43<16:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 33% ━━━━──────── 192/567 2.7s/it 8:46<16:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 193/567 2.7s/it 8:48<16:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 194/567 2.7s/it 8:51<16:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 195/567 2.7s/it 8:54<16:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 196/567 2.7s/it 8:56<16:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 197/567 2.7s/it 8:59<16:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 34% ━━━━──────── 198/567 2.7s/it 9:02<16:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 199/567 2.7s/it 9:05<16:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 200/567 2.7s/it 9:07<16:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 201/567 2.7s/it 9:10<16:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 202/567 2.7s/it 9:13<16:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 203/567 2.7s/it 9:15<16:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 35% ━━━━──────── 204/567 2.7s/it 9:18<16:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 205/567 2.7s/it 9:21<16:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 206/567 2.7s/it 9:24<16:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 207/567 2.7s/it 9:26<16:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 208/567 2.7s/it 9:29<16:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 36% ━━━━──────── 209/567 2.7s/it 9:32<16:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━──────── 210/567 2.7s/it 9:35<16:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━──────── 211/567 2.7s/it 9:37<16:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━──────── 212/567 2.7s/it 9:40<16:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━╸─────── 213/567 2.7s/it 9:43<16:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━╸─────── 214/567 2.7s/it 9:45<15:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 37% ━━━━╸─────── 215/567 2.7s/it 9:48<15:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 216/567 2.7s/it 9:51<15:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 217/567 2.7s/it 9:54<15:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 218/567 2.7s/it 9:56<15:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 219/567 2.7s/it 9:59<15:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 220/567 2.7s/it 10:02<15:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 38% ━━━━╸─────── 221/567 2.7s/it 10:04<15:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 39% ━━━━╸─────── 222/567 2.7s/it 10:07<15:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 39% ━━━━╸─────── 223/567 2.7s/it 10:10<15:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 39% ━━━━╸─────── 224/567 2.7s/it 10:13<15:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 39% ━━━━╸─────── 225/567 2.7s/it 10:15<15:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 39% ━━━━╸─────── 226/567 2.7s/it 10:18<15:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 227/567 2.7s/it 10:21<15:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 228/567 2.7s/it 10:23<15:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 229/567 2.7s/it 10:26<15:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 230/567 2.7s/it 10:29<15:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 231/567 2.7s/it 10:32<15:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 40% ━━━━╸─────── 232/567 2.7s/it 10:34<15:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━╸─────── 233/567 2.7s/it 10:37<15:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━╸─────── 234/567 2.7s/it 10:40<15:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━╸─────── 235/567 2.7s/it 10:42<14:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━╸─────── 236/567 2.7s/it 10:45<14:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━━─────── 237/567 2.7s/it 10:48<14:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 41% ━━━━━─────── 238/567 2.7s/it 10:51<14:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 42% ━━━━━─────── 239/567 2.7s/it 10:53<14:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 42% ━━━━━─────── 240/567 2.7s/it 10:56<14:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 42% ━━━━━─────── 241/567 2.7s/it 10:59<14:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 42% ━━━━━─────── 242/567 2.7s/it 11:01<14:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 42% ━━━━━─────── 243/567 2.7s/it 11:04<14:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 244/567 2.7s/it 11:07<14:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 245/567 2.7s/it 11:10<14:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 246/567 2.7s/it 11:12<14:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 247/567 2.7s/it 11:15<14:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 248/567 2.7s/it 11:18<14:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 43% ━━━━━─────── 249/567 2.7s/it 11:20<14:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 250/567 2.7s/it 11:23<14:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 251/567 2.7s/it 11:26<14:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 252/567 2.7s/it 11:28<14:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 253/567 2.7s/it 11:31<14:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 254/567 2.7s/it 11:34<14:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 44% ━━━━━─────── 255/567 2.7s/it 11:37<14:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 45% ━━━━━─────── 256/567 2.7s/it 11:39<14:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 45% ━━━━━─────── 257/567 2.7s/it 11:42<14:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 45% ━━━━━─────── 258/567 2.7s/it 11:45<14:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 45% ━━━━━─────── 259/567 2.7s/it 11:48<13:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 45% ━━━━━╸────── 260/567 2.7s/it 11:50<13:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 261/567 2.7s/it 11:53<13:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 262/567 2.7s/it 11:56<13:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 263/567 2.7s/it 11:58<13:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 264/567 2.7s/it 12:01<13:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 265/567 2.7s/it 12:04<13:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 46% ━━━━━╸────── 266/567 2.7s/it 12:07<13:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 267/567 2.7s/it 12:09<13:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 268/567 2.7s/it 12:12<13:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 269/567 2.7s/it 12:15<13:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 270/567 2.7s/it 12:17<13:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 271/567 2.7s/it 12:20<13:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 47% ━━━━━╸────── 272/567 2.7s/it 12:23<13:20

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 48% ━━━━━╸────── 273/567 2.7s/it 12:25<13:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 48% ━━━━━╸────── 274/567 2.7s/it 12:28<13:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 48% ━━━━━╸────── 275/567 2.7s/it 12:31<13:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 48% ━━━━━╸────── 276/567 2.8s/it 12:34<13:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 48% ━━━━━╸────── 277/567 2.8s/it 12:37<13:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 278/567 2.7s/it 12:39<13:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 279/567 2.7s/it 12:42<13:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 280/567 2.7s/it 12:45<13:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 281/567 2.7s/it 12:47<12:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 282/567 2.7s/it 12:50<12:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 49% ━━━━━╸────── 283/567 2.7s/it 12:53<12:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 284/567 2.7s/it 12:56<12:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 285/567 2.7s/it 12:58<12:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 286/567 2.7s/it 13:01<12:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 287/567 2.7s/it 13:04<12:41

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 288/567 2.7s/it 13:06<12:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 50% ━━━━━━────── 289/567 2.7s/it 13:09<12:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 51% ━━━━━━────── 290/567 2.7s/it 13:12<12:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 51% ━━━━━━────── 291/567 2.7s/it 13:15<12:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 51% ━━━━━━────── 292/567 2.7s/it 13:17<12:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 51% ━━━━━━────── 293/567 2.7s/it 13:20<12:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 51% ━━━━━━────── 294/567 2.7s/it 13:23<12:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 295/567 2.7s/it 13:26<12:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 296/567 2.7s/it 13:28<12:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 297/567 2.7s/it 13:31<12:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 298/567 2.7s/it 13:34<12:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 299/567 2.7s/it 13:37<12:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 52% ━━━━━━────── 300/567 2.7s/it 13:39<12:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 301/567 2.7s/it 13:42<12:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 302/567 2.7s/it 13:45<12:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 303/567 2.7s/it 13:47<12:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 304/567 2.7s/it 13:50<11:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 305/567 2.7s/it 13:53<11:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 53% ━━━━━━────── 306/567 2.7s/it 13:56<11:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 54% ━━━━━━────── 307/567 2.7s/it 13:58<11:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 54% ━━━━━━╸───── 308/567 2.7s/it 14:01<11:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 54% ━━━━━━╸───── 309/567 2.7s/it 14:04<11:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 54% ━━━━━━╸───── 310/567 2.7s/it 14:07<11:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 54% ━━━━━━╸───── 311/567 2.7s/it 14:09<11:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 312/567 2.7s/it 14:12<11:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 313/567 2.7s/it 14:15<11:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 314/567 2.8s/it 14:18<11:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 315/567 2.8s/it 14:20<11:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 316/567 2.7s/it 14:23<11:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 55% ━━━━━━╸───── 317/567 2.7s/it 14:26<11:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 318/567 2.7s/it 14:28<11:20

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 319/567 2.7s/it 14:31<11:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 320/567 2.7s/it 14:34<11:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 321/567 2.7s/it 14:37<11:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 322/567 2.7s/it 14:39<11:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 56% ━━━━━━╸───── 323/567 2.7s/it 14:42<11:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 57% ━━━━━━╸───── 324/567 2.7s/it 14:45<11:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 57% ━━━━━━╸───── 325/567 2.7s/it 14:48<10:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 57% ━━━━━━╸───── 326/567 2.7s/it 14:50<10:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 57% ━━━━━━╸───── 327/567 2.7s/it 14:53<10:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 57% ━━━━━━╸───── 328/567 2.7s/it 14:56<10:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━╸───── 329/567 2.7s/it 14:59<10:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━╸───── 330/567 2.7s/it 15:01<10:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━━───── 331/567 2.7s/it 15:04<10:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━━───── 332/567 2.7s/it 15:07<10:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━━───── 333/567 2.7s/it 15:09<10:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 58% ━━━━━━━───── 334/567 2.7s/it 15:12<10:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 335/567 2.7s/it 15:15<10:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 336/567 2.7s/it 15:18<10:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 337/567 2.7s/it 15:20<10:26

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 338/567 2.7s/it 15:23<10:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 339/567 2.7s/it 15:26<10:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 59% ━━━━━━━───── 340/567 2.7s/it 15:28<10:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 60% ━━━━━━━───── 341/567 2.7s/it 15:31<10:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 60% ━━━━━━━───── 342/567 2.7s/it 15:34<10:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 60% ━━━━━━━───── 343/567 2.7s/it 15:37<10:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 60% ━━━━━━━───── 344/567 2.7s/it 15:39<10:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 60% ━━━━━━━───── 345/567 2.7s/it 15:42<10:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 346/567 2.7s/it 15:45<10:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 347/567 2.7s/it 15:47<9:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 348/567 2.7s/it 15:50<9:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 349/567 2.7s/it 15:53<9:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 350/567 2.7s/it 15:56<9:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 61% ━━━━━━━───── 351/567 2.7s/it 15:58<9:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━───── 352/567 2.7s/it 16:01<9:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━───── 353/567 2.7s/it 16:04<9:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━───── 354/567 2.7s/it 16:07<9:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━╸──── 355/567 2.7s/it 16:09<9:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━╸──── 356/567 2.7s/it 16:12<9:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 62% ━━━━━━━╸──── 357/567 2.7s/it 16:15<9:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 358/567 2.7s/it 16:17<9:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 359/567 2.7s/it 16:20<9:26

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 360/567 2.7s/it 16:23<9:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 361/567 2.7s/it 16:26<9:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 63% ━━━━━━━╸──── 362/567 2.7s/it 16:28<9:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 363/567 2.7s/it 16:31<9:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 364/567 2.7s/it 16:34<9:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 365/567 2.7s/it 16:36<9:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 366/567 2.7s/it 16:39<9:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 367/567 2.7s/it 16:42<9:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 64% ━━━━━━━╸──── 368/567 2.7s/it 16:45<9:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 369/567 2.7s/it 16:47<9:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 370/567 2.7s/it 16:50<8:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 371/567 2.7s/it 16:53<8:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 372/567 2.7s/it 16:56<8:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 373/567 2.7s/it 16:58<8:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 65% ━━━━━━━╸──── 374/567 2.7s/it 17:01<8:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━╸──── 375/567 2.7s/it 17:04<8:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━╸──── 376/567 2.7s/it 17:06<8:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━╸──── 377/567 2.7s/it 17:09<8:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━━──── 378/567 2.7s/it 17:12<8:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 66% ━━━━━━━━──── 379/567 2.7s/it 17:15<8:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 380/567 2.7s/it 17:17<8:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 381/567 2.7s/it 17:20<8:26

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 382/567 2.7s/it 17:23<8:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 383/567 2.7s/it 17:26<8:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 384/567 2.7s/it 17:28<8:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 67% ━━━━━━━━──── 385/567 2.7s/it 17:31<8:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 386/567 2.7s/it 17:34<8:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 387/567 2.7s/it 17:37<8:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 388/567 2.7s/it 17:39<8:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 389/567 2.7s/it 17:42<8:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 390/567 2.7s/it 17:45<8:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 68% ━━━━━━━━──── 391/567 2.7s/it 17:47<8:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 392/567 2.7s/it 17:50<7:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 393/567 2.7s/it 17:53<7:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 394/567 2.7s/it 17:56<7:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 395/567 2.7s/it 17:58<7:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 69% ━━━━━━━━──── 396/567 2.7s/it 18:01<7:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━──── 397/567 2.7s/it 18:04<7:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━──── 398/567 2.7s/it 18:07<7:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━──── 399/567 2.7s/it 18:09<7:38

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━──── 400/567 2.7s/it 18:12<7:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━──── 401/567 2.7s/it 18:15<7:32

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 70% ━━━━━━━━╸─── 402/567 2.7s/it 18:17<7:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 403/567 2.7s/it 18:20<7:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 404/567 2.7s/it 18:23<7:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 405/567 2.7s/it 18:26<7:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 406/567 2.7s/it 18:28<7:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 407/567 2.7s/it 18:31<7:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 71% ━━━━━━━━╸─── 408/567 2.7s/it 18:34<7:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 409/567 2.8s/it 18:37<7:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 410/567 2.8s/it 18:39<7:16

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 411/567 2.8s/it 18:42<7:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 412/567 2.8s/it 18:45<7:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 72% ━━━━━━━━╸─── 413/567 2.8s/it 18:48<7:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 414/567 2.8s/it 18:51<7:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 415/567 2.8s/it 18:53<7:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 416/567 2.8s/it 18:56<6:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 417/567 2.7s/it 18:59<6:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 418/567 2.7s/it 19:02<6:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 73% ━━━━━━━━╸─── 419/567 2.7s/it 19:04<6:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 420/567 2.7s/it 19:07<6:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 421/567 2.7s/it 19:10<6:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 422/567 2.7s/it 19:12<6:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 423/567 2.7s/it 19:15<6:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 424/567 2.7s/it 19:18<6:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 74% ━━━━━━━━╸─── 425/567 2.7s/it 19:21<6:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 426/567 2.7s/it 19:23<6:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 427/567 2.7s/it 19:26<6:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 428/567 2.7s/it 19:29<6:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 429/567 2.7s/it 19:31<6:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 75% ━━━━━━━━━─── 430/567 2.7s/it 19:34<6:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 431/567 2.7s/it 19:37<6:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 432/567 2.7s/it 19:40<6:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 433/567 2.7s/it 19:42<6:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 434/567 2.7s/it 19:45<6:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 435/567 2.7s/it 19:48<5:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 76% ━━━━━━━━━─── 436/567 2.7s/it 19:51<5:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 437/567 2.7s/it 19:53<5:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 438/567 2.7s/it 19:56<5:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 439/567 2.7s/it 19:59<5:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 440/567 2.7s/it 20:01<5:45

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 441/567 2.7s/it 20:04<5:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 77% ━━━━━━━━━─── 442/567 2.7s/it 20:07<5:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 78% ━━━━━━━━━─── 443/567 2.7s/it 20:10<5:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 78% ━━━━━━━━━─── 444/567 2.7s/it 20:12<5:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 78% ━━━━━━━━━─── 445/567 2.7s/it 20:15<5:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 78% ━━━━━━━━━─── 446/567 2.7s/it 20:18<5:29

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 78% ━━━━━━━━━─── 447/567 2.7s/it 20:20<5:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━─── 448/567 2.7s/it 20:23<5:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━╸── 449/567 2.7s/it 20:26<5:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━╸── 450/567 2.7s/it 20:29<5:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━╸── 451/567 2.7s/it 20:31<5:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━╸── 452/567 2.7s/it 20:34<5:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 79% ━━━━━━━━━╸── 453/567 2.7s/it 20:37<5:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 454/567 2.7s/it 20:40<5:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 455/567 2.7s/it 20:42<5:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 456/567 2.7s/it 20:45<5:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 457/567 2.7s/it 20:48<5:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 458/567 2.7s/it 20:51<4:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 80% ━━━━━━━━━╸── 459/567 2.8s/it 20:53<4:57

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 81% ━━━━━━━━━╸── 460/567 2.7s/it 20:56<4:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 81% ━━━━━━━━━╸── 461/567 2.7s/it 20:59<4:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 81% ━━━━━━━━━╸── 462/567 2.8s/it 21:02<4:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 81% ━━━━━━━━━╸── 463/567 2.7s/it 21:04<4:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 81% ━━━━━━━━━╸── 464/567 2.7s/it 21:07<4:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 465/567 2.7s/it 21:10<4:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 466/567 2.7s/it 21:12<4:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 467/567 2.7s/it 21:15<4:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 468/567 2.7s/it 21:18<4:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 469/567 2.7s/it 21:21<4:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 82% ━━━━━━━━━╸── 470/567 2.7s/it 21:23<4:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━╸── 471/567 2.7s/it 21:26<4:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━╸── 472/567 2.7s/it 21:29<4:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━━── 473/567 2.7s/it 21:31<4:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━━── 474/567 2.7s/it 21:34<4:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━━── 475/567 2.7s/it 21:37<4:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 83% ━━━━━━━━━━── 476/567 2.7s/it 21:40<4:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 84% ━━━━━━━━━━── 477/567 2.7s/it 21:42<4:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 84% ━━━━━━━━━━── 478/567 2.7s/it 21:45<4:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 84% ━━━━━━━━━━── 479/567 2.7s/it 21:48<4:00

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 84% ━━━━━━━━━━── 480/567 2.7s/it 21:51<3:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 84% ━━━━━━━━━━── 481/567 2.7s/it 21:53<3:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 482/567 2.7s/it 21:56<3:52

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 483/567 2.7s/it 21:59<3:49

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 484/567 2.7s/it 22:02<3:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 485/567 2.7s/it 22:04<3:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 486/567 2.7s/it 22:07<3:41

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 85% ━━━━━━━━━━── 487/567 2.7s/it 22:10<3:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 488/567 2.7s/it 22:12<3:35

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 489/567 2.7s/it 22:15<3:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 490/567 2.7s/it 22:18<3:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 491/567 2.7s/it 22:21<3:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 492/567 2.7s/it 22:23<3:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 86% ━━━━━━━━━━── 493/567 2.7s/it 22:26<3:23

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 87% ━━━━━━━━━━── 494/567 2.7s/it 22:29<3:20

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 87% ━━━━━━━━━━── 495/567 2.7s/it 22:32<3:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 87% ━━━━━━━━━━── 496/567 2.7s/it 22:34<3:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 87% ━━━━━━━━━━╸─ 497/567 2.7s/it 22:37<3:12

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 87% ━━━━━━━━━━╸─ 498/567 2.7s/it 22:40<3:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 499/567 2.8s/it 22:43<3:08

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 500/567 2.8s/it 22:45<3:05

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 501/567 2.8s/it 22:48<3:02

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 502/567 2.8s/it 22:51<2:59

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 503/567 2.8s/it 22:54<2:56

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 88% ━━━━━━━━━━╸─ 504/567 2.8s/it 22:57<2:54

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 505/567 2.8s/it 22:59<2:51

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 506/567 2.8s/it 23:02<2:48

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 507/567 2.8s/it 23:05<2:46

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 508/567 2.8s/it 23:08<2:43

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 509/567 2.8s/it 23:10<2:40

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 89% ━━━━━━━━━━╸─ 510/567 2.8s/it 23:13<2:41

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 90% ━━━━━━━━━━╸─ 511/567 2.8s/it 23:16<2:37

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 90% ━━━━━━━━━━╸─ 512/567 2.8s/it 23:19<2:34

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 90% ━━━━━━━━━━╸─ 513/567 2.8s/it 23:22<2:31

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 90% ━━━━━━━━━━╸─ 514/567 2.8s/it 23:24<2:27

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 90% ━━━━━━━━━━╸─ 515/567 2.8s/it 23:27<2:24

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━╸─ 516/567 2.8s/it 23:30<2:21

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━╸─ 517/567 2.8s/it 23:33<2:18

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━╸─ 518/567 2.8s/it 23:35<2:15

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━╸─ 519/567 2.8s/it 23:38<2:13

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━━─ 520/567 2.8s/it 23:41<2:10

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 91% ━━━━━━━━━━━─ 521/567 2.8s/it 23:44<2:07

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 522/567 2.8s/it 23:46<2:04

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 523/567 2.7s/it 23:49<2:01

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 524/567 2.7s/it 23:52<1:58

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 525/567 2.7s/it 23:55<1:55

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 526/567 2.8s/it 23:57<1:53

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 92% ━━━━━━━━━━━─ 527/567 2.8s/it 24:00<1:50

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 93% ━━━━━━━━━━━─ 528/567 2.8s/it 24:03<1:47

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 93% ━━━━━━━━━━━─ 529/567 2.7s/it 24:06<1:44

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 93% ━━━━━━━━━━━─ 530/567 2.7s/it 24:08<1:42

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 93% ━━━━━━━━━━━─ 531/567 2.7s/it 24:11<1:39

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 93% ━━━━━━━━━━━─ 532/567 2.7s/it 24:14<1:36

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 533/567 2.7s/it 24:17<1:33

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 534/567 2.7s/it 24:19<1:30

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 535/567 2.7s/it 24:22<1:28

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 536/567 2.7s/it 24:25<1:25

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 537/567 2.7s/it 24:28<1:22

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 94% ━━━━━━━━━━━─ 538/567 2.7s/it 24:30<1:19

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━─ 539/567 2.7s/it 24:33<1:17

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━─ 540/567 2.7s/it 24:36<1:14

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━─ 541/567 2.7s/it 24:39<1:11

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━─ 542/567 2.7s/it 24:41<1:09

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━─ 543/567 2.7s/it 24:44<1:06

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 95% ━━━━━━━━━━━╸ 544/567 2.7s/it 24:47<1:03

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 96% ━━━━━━━━━━━╸ 545/567 2.7s/it 24:50<1:00

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 96% ━━━━━━━━━━━╸ 546/567 2.7s/it 24:52<57.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 96% ━━━━━━━━━━━╸ 547/567 2.7s/it 24:55<54.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 96% ━━━━━━━━━━━╸ 548/567 2.7s/it 24:58<52.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 96% ━━━━━━━━━━━╸ 549/567 2.7s/it 25:00<49.2s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 550/567 2.7s/it 25:03<46.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 551/567 2.7s/it 25:06<43.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 552/567 2.7s/it 25:09<41.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 553/567 2.7s/it 25:11<38.3s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 554/567 2.7s/it 25:14<35.6s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 97% ━━━━━━━━━━━╸ 555/567 2.7s/it 25:17<32.8s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 556/567 2.7s/it 25:20<30.1s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 557/567 2.7s/it 25:22<27.3s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 558/567 2.7s/it 25:25<24.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 559/567 2.7s/it 25:28<21.8s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 560/567 2.7s/it 25:31<19.2s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 98% ━━━━━━━━━━━╸ 561/567 2.7s/it 25:33<16.4s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 99% ━━━━━━━━━━━╸ 562/567 2.8s/it 25:36<13.8s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 99% ━━━━━━━━━━━╸ 563/567 2.8s/it 25:39<11.0s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 99% ━━━━━━━━━━━╸ 564/567 2.7s/it 25:42<8.2s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 99% ━━━━━━━━━━━╸ 565/567 2.7s/it 25:44<5.5s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 99% ━━━━━━━━━━━╸ 566/567 2.7s/it 25:47<2.7s

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 567/567 2.7s/it 25:50

                   all        567       4129      0.795      0.765      0.846      0.679


Speed: 6.0ms preprocess, 2708.2ms inference, 0.0ms loss, 0.1ms postprocess per image


Results saved to /home/tj_students/leaf-localisation-main/deployment/runs/detect/val-27


Precision  : 0.795
Recall     : 0.765
mAP50      : 0.846
mAP50-95   : 0.679


In [8]:
from pathlib import Path
Path("compiler_logs").mkdir(parents=True, exist_ok=True)
# The two cells below are ONLY needed if you uncommented the full-INT8 Coral export above
# and want to compile models/best_m_int8.tflite for the physical Edge TPU chip.
# Requires the Coral compiler installed once via sudo (no passwordless sudo on this machine):
#   curl -fsSL https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo gpg --dearmor -o /etc/apt/keyrings/google.gpg
#   echo "deb [signed-by=/etc/apt/keyrings/google.gpg] https://packages.cloud.google.com/apt coral-edgetpu-stable main" | sudo tee /etc/apt/sources.list.d/coral-edgetpu.list
#   sudo apt-get update && sudo apt-get install edgetpu-compiler

In [9]:
import shutil
from pathlib import Path

int8_full_path = Path("models/best_m_int8.tflite")
if not int8_full_path.exists():
    print("Skipping Edge TPU compile: models/best_m_int8.tflite not found — "
          "uncomment the full-INT8 export cell above to generate it first.")
elif shutil.which("edgetpu_compiler") is None:
    print("Skipping Edge TPU compile: edgetpu_compiler is not installed on this machine. "
          "See the setup commands in the cell above (requires sudo).")
else:
    !edgetpu_compiler --out_dir compiler_logs "{int8_full_path}" | tee compiler_logs/edgetpu_log.txt


Skipping Edge TPU compile: edgetpu_compiler is not installed on this machine. See the setup commands in the cell above (requires sudo).


In [10]:
# NCNN export currently fails on this model: Ultralytics 8.4.104's NCNN exporter
# disables the end2end NMS wrapper (`NMSModel`) for this format, but the Detect head's
# forward pass still branches on `self.end2end` and expects dict-style preds with
# 'feats'/'one2one' keys — so the dry-run forward pass inside export() crashes with
# `KeyError: 'feats'`. This is a genuine YOLO26 + NCNN-exporter incompatibility in this
# Ultralytics version, not a config issue — same category of problem as the onnx2tf hang
# documented above. Exported here defensively so a fix/version bump doesn't require
# rewriting this cell; ncnn_path stays None if export isn't yet supported.
try:
    ncnn_path = model.export(format="ncnn", imgsz=1280)
    print("NCNN model:", ncnn_path)
except Exception as e:
    ncnn_path = None
    print(f"NCNN export failed and is being skipped: {type(e).__name__}: {e}")


Ultralytics 8.4.104 🚀 Python-3.10.12 torch-2.13.0+cu130 CPU (AMD EPYC 7313 16-Core Processor)


WARNING ⚠️ NCNN export does not support end2end models, disabling end2end branch.


NCNN export failed and is being skipped: KeyError: 'feats'


In [11]:
if ncnn_path is not None:
    ncnn_metrics = YOLO(ncnn_path).val(data="../data.yaml", imgsz=1280)
    print(f"Precision  : {ncnn_metrics.box.mp:.3f}")
    print(f"Recall     : {ncnn_metrics.box.mr:.3f}")
    print(f"mAP50      : {ncnn_metrics.box.map50:.3f}")
    print(f"mAP50-95   : {ncnn_metrics.box.map:.3f}")
else:
    ncnn_metrics = None
    print("Skipping NCNN validation: export failed above.")


Skipping NCNN validation: export failed above.


In [12]:
import pandas as pd
from pathlib import Path

rows = [
    {"format": "pt (baseline)", "precision": metrics.box.mp, "recall": metrics.box.mr,
     "mAP50": metrics.box.map50, "mAP50-95": metrics.box.map},
    {"format": "int8 tflite", "precision": int8_metrics.box.mp, "recall": int8_metrics.box.mr,
     "mAP50": int8_metrics.box.map50, "mAP50-95": int8_metrics.box.map},
]
if ncnn_metrics is not None:
    rows.append({"format": "ncnn", "precision": ncnn_metrics.box.mp, "recall": ncnn_metrics.box.mr,
                  "mAP50": ncnn_metrics.box.map50, "mAP50-95": ncnn_metrics.box.map})
df = pd.DataFrame(rows)
Path("reports").mkdir(parents=True, exist_ok=True)
df.to_csv("reports/deployment_metrics.csv", index=False)
df


,format,precision,recall,mAP50,mAP50-95
0,pt (baseline),0.803255,0.771858,0.854046,0.688416
1,int8 tflite,0.795230,0.765076,0.846032,0.679071
